test/test_suite_13.py

```diff
@@ -2059,3 +2063,128 @@ def test_create_engine_wo_database(self):
         engine = create_engine(get_db_url().split("/database")[0])
         with engine.connect() as connection:
             assert connection.connection.database is None
+
+
+@pytest.mark.skipif(
+    bool(os.environ.get("SPANNER_EMULATOR_HOST")), reason="Skipped on emulator"
+)
+class SequenceTest(_SequenceTest):
+    @classmethod
+    def define_tables(cls, metadata):
+        Table(
+            "seq_pk",
+            metadata,
+            Column(
+                "id",
+                Integer,
+                sqlalchemy.Sequence("tab_id_seq"),
+                primary_key=True,
+            ),
+            Column("data", String(50)),
+        )
+
+        Table(
+            "seq_opt_pk",
+            metadata,
+            Column(
+                "id",
+                Integer,
+                sqlalchemy.Sequence("tab_id_seq_opt", data_type=Integer, optional=True),
+                primary_key=True,
+            ),
+            Column("data", String(50)),
+        )
+
+        Table(
+            "seq_no_returning",
+            metadata,
+            Column(
+                "id",
+                Integer,
+                sqlalchemy.Sequence("noret_id_seq"),
+                primary_key=True,
+            ),
+            Column("data", String(50)),
+            implicit_returning=False,
+        )
+
+    def test_insert_roundtrip(self, connection):
+        connection.execute(self.tables.seq_pk.insert(), dict(data="some data"))
+        self._assert_round_trip(self.tables.seq_pk, connection)
+
+    def test_insert_lastrowid(self, connection):
+        r = connection.execute(self.tables.seq_pk.insert(), dict(data="some data"))
+        assert len(r.inserted_primary_key) == 1
+        is_instance_of(r.inserted_primary_key[0], int)
+
+    def test_nextval_direct(self, connection):
+        r = connection.execute(self.tables.seq_pk.c.id.default)
+        is_instance_of(r, int)
```

Comment:
Why are we not checking for equal value as expected here and only type?

Reply:
The actual test validates `eq_(r, testing.db.dialect.default_sequence_base)`, where `testing.db.dialect.default_sequence_base` is the starting value and is normally fixed (for ex: `1` in most dialects).

In Spanner SQLAlchemy, we support sequences through bit reversal and hence this is never fixed and keep changing.
This restriction does not allow us to verify the value and instead verify if the returned values is an `int`.

Reply:
Can we verify that the values are incremental atleast, or something of those sorts?

Reply:
As discussed offline, there is no other alternative due to randomness.